app/src/main/java/com/framgia/fsalon/screen/info/InfoUserViewModel.java

```diff
@@ -0,0 +1,107 @@
+package com.framgia.fsalon.screen.info;
+
+import android.content.Intent;
+import android.databinding.BaseObservable;
+import android.databinding.Bindable;
+import android.support.v4.app.Fragment;
+import android.view.View;
+
+import com.framgia.fsalon.BR;
+import com.framgia.fsalon.screen.login.LoginActivity;
+import com.framgia.fsalon.utils.navigator.Navigator;
+
+/**
+ * Exposes the data to be used in the InfoUser screen.
+ */
+public class InfoUserViewModel extends BaseObservable implements InfoUserContract.ViewModel {
+    private InfoUserContract.Presenter mPresenter;
+    private int mProgressBarVisibility;
+    private boolean mIsSkipLogin;
+    private Fragment mFragment;
+    private Navigator mNavigator;
+
+    public InfoUserViewModel(Fragment fragment) {
+        mNavigator = new Navigator(fragment);
+        mIsSkipLogin = true;
+        setProgressBarVisibility(View.GONE);
+        mFragment = fragment;
+    }
+
+    @Bindable
+    public boolean isSkipLogin() {
+        return mIsSkipLogin;
+    }
+
+    public void setSkipLogin(boolean skipLogin) {
+        mIsSkipLogin = skipLogin;
+        notifyPropertyChanged(BR.skipLogin);
+    }
+
+    @Bindable
+    public int getProgressBarVisibility() {
+        return mProgressBarVisibility;
+    }
+
+    @Override
+    public void onStart() {
+        mPresenter.onStart();
+    }
+
+    @Override
+    public void onStop() {
+        mPresenter.onStop();
+    }
+
+    @Override
+    public void setPresenter(InfoUserContract.Presenter presenter) {
+        mPresenter = presenter;
+    }
+
+    public void setProgressBarVisibility(int progressBarVisibility) {
+        mProgressBarVisibility = progressBarVisibility;
+        notifyPropertyChanged(BR.progressBarVisibility);
+    }
+
+    public void sentToButtonLogin() {
+        setSkipLogin(true);
+    }
+
+    @Override
+    public void sentToButtonLogout() {
+        setSkipLogin(false);
+    }
+
+    @Override
+    public void onClickButton() {
+        if (!isSkipLogin()) {
+            mPresenter.logout();
+        } else {
+            Intent intent = new Intent(mFragment.getContext(), LoginActivity.class);
+            mNavigator.startActivity(intent);
+            mNavigator.finishActivity();
+        }
+    }
+
+    @Override
+    public void showProgressbar() {
+        setProgressBarVisibility(View.VISIBLE);
+    }
+
+    @Override
+    public void hideProgressbar() {
+        setProgressBarVisibility(View.GONE);
+    }
+
+    @Override
+    public void onLogoutError(String message) {
+        mNavigator.showToast(message);
+    }
+
+    @Override
+    public void onLogoutSuccess(String message) {
+        mNavigator.showToast(message);
+        Intent intent = new Intent(mFragment.getContext(), LoginActivity.class);
```

Comment:
pls use mNavigator.getContext instead of mFragment.getContext()
and remove mFragment instance variable